miden-lib/src/transaction/inputs.rs

```diff
@@ -245,8 +244,11 @@ fn add_account_to_advice_inputs(
     // --- account code -------------------------------------------------------
     let code = account.code();
 
-    // extend the merkle store with account code tree
-    inputs.extend_merkle_store(code.procedure_tree().inner_nodes());
+    // extend the advice_map with the account code data and procedure length
+    let num_procs = code.as_elements().len() / 8;
+    let mut procs = code.as_elements();
+    procs.insert(0, Felt::from(num_procs as u32));
+    inputs.extend_map([(*code.procedure_commitment(), procs)]);
```

Comment:
I think the term "procedure length" is a bit ambiguous. Much better to use "number of procedures" and `num_procs` (we should do this in other places too).

Also, I would try to centralize the constants somewhere. For example, instead of using `8`, we could use something like `AccountProcedure::NUM_ELEMENTS_PER_PROC`.

Reply:
Added.